sway-error/src/warning.rs

```diff
@@ -84,7 +84,14 @@ pub enum Warning {
     DeadStorageDeclarationForFunction {
         unneeded_attrib: String,
     },
-    MatchExpressionUnreachableArm,
+    MatchExpressionUnreachableArm {
+        match_value: Span,
+        preceding_arms: Span,
```

Comment:
Should this be `Vec<Span>`?

Reply:
It could be I also thought of it. But since the overall span is used in a few messages I wanted to spare the (one day hopefully macro-based) diagnostic definition of calling `Span::join_all()` every time without having any other benefit of knowing every part of the joint span.

That was the reason I placed the "burden" of getting the joint span to the client.

Interesting question will be how similar situations are handled in general throughout the errors/warnings. Perhaps we come up with a different general guidelines, but so far I would leave it as it is.